client/site-profiler/components/site-profiler.tsx

```diff
@@ -32,7 +32,8 @@ export default function SiteProfiler() {
 	const updateDomainQueryParam = ( value: string ) => {
 		// Update the domain query param;
 		// URL param is the source of truth
-		queryParams.set( 'domain', value );
+		value ? queryParams.set( 'domain', value ) : queryParams.delete( 'domain' );
```

Comment:
If a value is an empty string, delete query param to avoid situation like `?domain=`